src/puppetlabs/trapperkeeper/authorization/ring_middleware.clj

```diff
@@ -242,15 +248,25 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Public
 
+(schema/defn authorization-check :- AuthResultWithRequest
+  "A function that checks that the request is allowed by the provided rules.
+  Returns a map with the request with auth info added, whether the request is
+  authorized, and a message."
+  [request :- ring/Request
+   rules :- rules/Rules
+   allow-header-cert-info :- schema/Bool]
+  (let [req (add-authinfo request allow-header-cert-info)
+        name (ring/authorized-name req)]
+     (assoc (rules/allowed? rules req name) :req req)))
```

Comment:
Currently my implementation of getting this running with a servlet doesn't use the value of `:req` at all. I'm not sure whether it's really necessary to have it in what this function returns, but it could potentially be useful and it makes it easier for `wrap-authorization-check` to just use this function.